script.js

```diff
@@ -0,0 +1,298 @@
+      if (keyName === 'Shift') { this.toggleCapsLock(); }
+    });
+    this.elements.keysContainer.addEventListener('click', (event) => {
+      if (event.target !== this.elements.keysContainer) {
+        const key = event.target;
+        const index = this.elements.keys.indexOf(key);
+        const caps = this.properties.capsLock ? 'caps' : 'nocaps';
+        const keyName = this.layouts[this.properties.lang][caps][index];
+        if (keyName !== 'Caps Lock') { key.classList.add('keyboard--key-active'); }
+        switch (keyName) {
+          case 'Backspace':
+            this.properties.value = this.properties.value.slice(0, -1);
+            break;
+          case 'Enter':
+            this.properties.value += '\n';
+            break;
+          case 'Caps Lock':
+            key.classList.toggle('keyboard--key-active');
+            this.toggleCapsLock();
+            break;
+          case 'Shift':
+            break;
+          case 'Del':
+            this.properties.value = this.properties.value.slice(1);
+            break;
+          case 'Tab':
+            this.properties.value += '\t';
+            break;
+          case 'Ctrl':
+            break;
+          case 'Alt':
+            break;
+          case 'Win':
+            break;
+          default:
+            this.properties.value += keyName;
+            break;
+        }
+        if (keyName !== 'Caps Lock') { setTimeout(() => { key.classList.remove('keyboard--key-active'); }, 300); }
+        this.triggerEvent('oninput');
+      }
+    });
+  }
+
+  createKeys() {
+    this.hardKeys.forEach((keyName) => {
+      const key = document.createElement('div');
+      key.classList.add('keyboard--key');
+      this.elements.keys.push(key);
+      switch (keyName) {
+        case 'Backspace':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'Enter':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'CapsLock':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftLeft':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftRight':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'Delete':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'Tab':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'ControlLeft':
+          break;
+        case 'ControlRight':
+          break;
+        case 'Space':
+          key.classList.add('keyboard--key-widest');
+          break;
+        case 'AltLeft':
+          break;
+        case 'AltRight':
+          break;
+        case 'ArrowLeft':
+          break;
+        case 'ArrowRight':
+          break;
+        case 'ArrowUp':
+          break;
+        case 'ArrowDown':
+          break;
+        case 'MetaLeft':
+          break;
```

Comment:
This also can be refactored to:
`        case 'AltLeft':
        case 'AltRight':
        case 'ArrowLeft':
        case 'ArrowRight':
        case 'ArrowUp':
        case 'ArrowDown':
        case 'MetaLeft':`